metrics/resource_view_test.go

```diff
@@ -166,6 +166,90 @@ func TestResourceAsString(t *testing.T) {
 	}
 }
 
+func TestRemoveInactiveMeter(t *testing.T) {
+	testNow := time.Date(2050, time.August, 7, 9, 0, 0, 0, time.UTC)
+	timeNow = func() time.Time { return testNow }
+	testCases := []struct {
+		name              string
+		metersForTest     *meters
+		inactiveThreshold time.Duration
+		remainingMeters   int
+	}{
+		{
+			name: "all inactive",
+			metersForTest: &meters{
+				meters: map[string]*meterExporter{
+					"": {m: &defaultMeterImpl{}, t: timeNow()},
+					// using defaulteMeterImple as we don't want to add real meters to global states.
+					"foo": {m: &defaultMeterImpl{}, t: timeNow().Add(-2 * time.Hour)},
+					"bar": {m: &defaultMeterImpl{}, t: timeNow().Add(-time.Hour - time.Microsecond)},
+				}},
+			inactiveThreshold: time.Hour,
+			remainingMeters:   1, // we don't delete the default one
+		},
+		{
+			name: "all active",
+			metersForTest: &meters{
+				meters: map[string]*meterExporter{
+					"":    {m: &defaultMeterImpl{}, t: timeNow()},
+					"foo": {m: &defaultMeterImpl{}, t: timeNow().Add(-30 * time.Minute)},
+					"bar": {m: &defaultMeterImpl{}, t: timeNow().Add(-time.Hour + time.Millisecond)},
+				}},
+			inactiveThreshold: time.Hour,
+			remainingMeters:   3,
+		},
+		{
+			name: "With both active and inactive meters",
+			metersForTest: &meters{
+				meters: map[string]*meterExporter{
+					"":    {m: &defaultMeterImpl{}, t: timeNow()},
+					"foo": {m: &defaultMeterImpl{}, t: timeNow().Add(-2 * time.Hour)},
+					"bar": {m: &defaultMeterImpl{}, t: timeNow().Add(-30 * time.Minute)},
+				}},
+			inactiveThreshold: time.Hour,
+			remainingMeters:   2,
+		},
+	}
+
+	for _, tc := range testCases {
+		removeInactiveMeters(tc.inactiveThreshold, tc.metersForTest)
+		if len(tc.metersForTest.meters) != tc.remainingMeters {
+			t.Fatalf("Expect %d remaining meters, but got %d", tc.remainingMeters, len(tc.metersForTest.meters))
+		}
+	}
+	timeNow = time.Now
```

Comment:
This line won't be reached if `t.Fatalf` happens before. I think we want to use a defer.

Reply:
Thanks.  Still trying to get better with golang :-)